xla/service/gpu/autotuning/gemm_fusion_autotuner.cc

```diff
@@ -833,24 +833,32 @@ GemmFusionAutotunerImpl::CompileAll(AutotunerCompileUtil& compile_util,
       VLOG(10) << "Dumping fusion computation: "
                << fusion->called_computation()->ToString();
       for (const Config& config : gemm_config_set) {
-        thread_pool_->Schedule([&, fusion] {
-          VLOG(10) << "Trying configuration forceable through: "
-                      "--xla_gpu_override_gemm_autotuner='"
-                   << Serialize(config) << "'";
-          VLOG(10) << "WARNING: you are running in multithreaded-mode, the "
-                      "last configuration printed out might not be the one "
-                      "causing issues! Use "
-                      "--xla_gpu_force_compilation_parallelism=1 to fix.";
-          absl::StatusOr<bool> has_executable =
-              compile(fusion, config, gemm_config_set.size() > 1);
-          TF_CHECK_OK(has_executable.status())
-              << "Failure occured when compiling fusion " << fusion->name()
-              << " with config '" << ToString(config)
-              << "'\nFused HLO computation:\n"
-              << fusion->fused_instructions_computation()->ToString();
-          log(has_executable.value());
-          counter.DecrementCount();
-        });
+        thread_pool_->Schedule(
+            [&, fusion] {
+              VLOG(10) << "Trying configuration forceable through: "
+                          "--xla_gpu_override_gemm_autotuner='"
+                       << Serialize(config) << "'";
+              VLOG(10) << "WARNING: you are running in multithreaded-mode, the "
+                          "last configuration printed out might not be the one "
+                          "causing issues! Use "
+                          "--xla_gpu_force_compilation_parallelism=1 to fix.";
+              absl::StatusOr<bool> has_executable =
+                  compile(fusion, config, gemm_config_set.size() > 1);
+              if (has_executable.ok()) {
+                log(has_executable.value());
+              } else {
+                log(/*success=*/false);
+                LOG(WARNING)
+                    << "Failure occured when compiling fusion "
+                    << fusion->name() << " with config '" << ToString(config)
+                    << "'\nFused HLO computation:\n"
+                    << fusion->fused_instructions_computation()->ToString();
+                if (!debug_options_.xla_gpu_exhaustive_tiling_search()) {
```

Comment:
Wait are you saying we still CHECK-fail otherwise? I definitely have seen cases where we fail to compile some of the tilings, but the execution continues. Or is it another check?

Forking the logic like that indeed does not seem ideal. (neither is crash-checking, i guess it could update a shared status with a mutex)

Reply:
> Wait are you saying we still CHECK-fail otherwise?

Yes.

Reply:
I'm still a bit hesitant about adding a conditional there, seems weird to only not-crash in exhaustive mode, what stops the non-exhaustive mode from crashing. I'm really confused though how I was seeing those as log messages and not as crashes.

Reply:
> what stops the non-exhaustive mode from crashing

Probably that all the fixed configs used in the non-exhaustive mode are rather sane.

Reply:
> I'm still a bit hesitant

How do you suggest to fix the problem then?

Reply:
@vwbaker suggests that there's already some magic to skip such tilings, does it pass if you update the regex here to include your error? https://github.com/openxla/xla/blob/main/xla/stream_executor/cuda/cuda_asm_compiler.cc#L385

Reply:
yes, that works

Reply:
do you want to change the PR to do that? You probably have newer/internal ptxas?

Reply:
> do you want to change the PR to do that? 

OK

> You probably have newer/internal ptxas?

Nothing special - Cuda compilation tools, release 12.5, V12.5.40, Build cuda_12.5.r12.5/compiler.34177558_0